### PR TITLE
Fix Whisper CI

### PR DESCRIPTION
**Pull request by ydshieh**
Finally CI is green

### CODE REVIEW

File: src/transformers/generation/flax_utils.py
Comment (ydshieh) [inline src/transformers/generation/flax_utils.py:405]: Diverge a bit from torch one, but let's try to match them in a follow-up when we feel it's time to do it.
Here is just to make CI green

**Comment by ydshieh** [comment]
> tests/models/rt_detr/test_image_processing_rt_detr.py::RtDetrImageProcessingTest::test_fast_is_faster_than_slow

it's flaky

**Comment by LysandreJik** [comment]
Should this be merged?

**Comment by ydshieh** [comment]
> Should this be merged?

Yes, but I am waiting for an approval. Currently, `whisper` has ~170 failed tests without this PR.

File: src/transformers/generation/utils.py
Comment (ArthurZucker) [inline src/transformers/generation/utils.py:1459]: thanks for putting this back!

**Review by ArthurZucker — APPROVED** [review]
Thanks! 🤗

**Comment by gante** [comment]
Hey guys 👋 @ydshieh @ArthurZucker 

The generate changes in this PR are incorrect (changing `max_length`) :) I'm reverting them as part of https://github.com/huggingface/transformers/pull/34807, but I'd recommend rolling back this PR.

If Whisper's tests are failing, then it is probably something wrong with either whisper or whisper's tests

**Comment by ydshieh** [comment]
Hi @gante This PR is only added

> if generation_config.max_length == GenerationConfig().max_length

to avoid the failure caused in #34377 (which was an attempt fix for #34026)

> The generate changes in this PR are incorrect

I believe the logic is wrong already in the previous PR #34377  and or #34026, right?

**Comment by gante** [comment]
Yeah I think it's from the older pipeline PR 😭 (`generation_config.max_length = generation_config.max_length + input_ids_seq_length` shouldn't exist)

**Comment by gante** [comment]
But the check for `max_position_embeddings` should exist either -- some models can generate beyond this value (e.g. llama)

**Comment by ydshieh** [comment]
So if we want to rollback, it's more involved (i.e. not revert this PR but some previous changes too).
In any case, ping me before a new change on this PR so I can see if the Whisper CI is still OK 🙏 .

**Comment by poedator** [comment]
This PR messes up generation tests. I noticed this when working on GPT2Model PR #35761.
looking forward to merging of https://github.com/huggingface/transformers/pull/34807 or some other solution

**Pull request merged**
